firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -301,6 +331,76 @@ public Query whereArrayContains(@NonNull FieldPath fieldPath, @NonNull Object va
     return whereHelper(fieldPath, Operator.ARRAY_CONTAINS, value);
   }
 
+  /**
+   * Creates and returns a new Query with the additional filter that documents must contain the
+   * specified field, the value must be an array, and that the array must contain at least one value
+   * from the provided array.
+   *
+   * <p>A Query can have only one whereArrayContainsAny() filter and it cannot be combined with
+   * whereArrayContains() or whereIn().
+   *
+   * @param field The name of the field containing an array to search.
+   * @param value The array that contains the values to match.
+   * @return The created Query.
+   */
+  @NonNull
+  @PublicApi
+  public Query whereArrayContainsAny(@NonNull String field, @NonNull List<Object> value) {
```

Comment:
We need to remove `@PublicApi` and `public` before checking this in (and add a TODO(in-queries) to expose).